### ISSUE TITLE
img elements with no alt attribute should maintain a role of img

### ISSUE THREAD

**Issue by ckundo** [eps1lon/dom-accessibility-api#429]
Currently, `getRole` on an image that is missing an `alt` attribute returns `null`. this is incorrect.

given the following markup:

```js
const imgEl = (new JSDOM(<img src='logo.png"'>`)).window.document.body.querySelector('img')
```

`getRole(imgEl)` should return `img`, but instead it returns `null`. (when adding an `alt` attribute, `getRole` return `img` correctly).

A completely missing `alt` attribute is notably different from `alt=""` or simply `alt`;  adding an `alt` attribute with an empty value _should_ remove the node from the AOM, whereas omitting it should _not_. using an empty value is a way for authors to instruct the browser to ignore an image.

Confirmed the above expected behavior in Chrome.

[related to #428]

**Comment by eps1lon** [comment]
Actually it should return `presentation`. By spec "img (alt attribute value is an empty string, i.e. alt="" or alt with no value in the markup)" maps to `role="presentation"` (see https://w3c.github.io/html-aam/#el-img-empty-alt).

**Comment by eps1lon** [comment]
Exception for https://github.com/eps1lon/dom-accessibility-api/issues/429#issuecomment-695908776: They should only have `role="presentation"` if they don't have any global ARIA attributes (following https://rawgit.com/w3c/aria/stable/#conflict_resolution_presentation_none)

**Comment by ckundo** [comment]
@eps1lon thanks for the response.

>  alt="" or alt with no value

This is different than a _missing_ alt attribute. many images on the web are missing an alt attribute, and marking these as presentational would hide images from assistive tech - screen readers, voice control, anything that uses accessible name. in these cases, it's almost certainly not the intent of the author to hide the images by omitting an `alt` attribute.

all current browsers behave how I'm describing: omitting alt _attribute_ maintains role of `img`, adding an empty string or no value removes the image from the AOM by marking it presentational.

**Comment by eps1lon** [comment]
> This is different than a missing alt attribute. many images on the web are missing an alt attribute, and marking these as presentational would hide images from assistive tech - screen readers, voice control, anything that uses accessible name. in these cases, it's almost certainly not the intent of the author to hide the images by omitting an alt attribute.

Sure, I'd suggest [opening an issue on w3c/html-aam](https://github.com/w3c/html-aam/issues/new) since I'm not a spec editor. 

From what I learned images with `alt=""` are considered decorative and hiding them from AT only makes sense to me. But maybe you have another use case in mind so asking spec folks is probably more productive.

**Comment by ckundo** [comment]
talked with @jnurthen about it already, he confirmed this: `alt=""` removes the image from the AOM. `alt="anything"` includes it, naturally. _no alt attribute_ keeps the image in the AOM, since it's still an image.

**Comment by eps1lon** [comment]
> no alt attribute keeps the image in the AOM, since it's still an image.

I'm not sure why you keep saying this. `role="presentation"` does not hide them. Unless I'm missunderstand what "alt with no value" means. Since they specify both "empty" and "no value" I'd assume "no value" means `image.alt = null` which is missing attribute. Could you clarify how one produces 

1. `alt=""`
2. `alt` with no value
3. missing `alt`
?

**Comment by ckundo** [comment]
ah, I think the confusion is this:

`<img />` is definitely not the same as `<img alt>`, or `<img alt="">`. the last two are equivalent.

**Comment by ckundo** [comment]
> Three common uses of role presentation are:
> 
> Hiding a decorative image; it is equivalent to giving the image null alt text.

https://www.w3.org/TR/wai-aria-practices-1.1/#presentation_role

this is what i was describing about the `presentation` role.

**Comment by eps1lon** [comment]
> is definitely not the same as <img alt>, or <img alt="">. the last two are equivalent.

I don't know how one would distinguish them with the DOM API which is why I don't think the spec talks about both cases. To me

1. `alt=""` means `<img alt />` **and** `<img alt="" />`
   `image.getAttribute('alt')` returns `""` for both cases
2. "`alt` with no value" means `<img />`
   `img.hasAttribute('alt')` returns `false` for this case

**Comment by eps1lon** [comment]
Ok so I think html-aam might be misleading. If you just focus on  "`alt` attribute value is an empty string" then  it would indeed not include `<img />` because the attribute value would be `null` for `<img />`. 

As far as I can tell "`alt` with no value in the markup" should just be removed since you have no way to distinguish that without literally looking at the markup. By html living standard `<img alt />` and `<img alt="" />` are equivalent.

**Comment by ckundo** [comment]
@eps1lon yep, agreed. i can update #430 to return `presentation` instead of `null`, and I think otherwise it's correct..

**Comment by ckundo** [comment]
case in point, here's how Chrome exposes elements per `getComputedAccessibleNode`

```html
<!-- Chrome browser getComputedAccessibleNode(img) results -->

<img alt="something" /> <!-- role: image, name: "something" -->
<img /> <!-- role: image, name: null -->

<!-- equivalent: role: null, name: null -->
<img alt />
<img alt="" />
<img role="presentation" />
```

https://codepen.io/ckundo/pen/oNxQdPO?editors=1111

**Comment by eps1lon** [comment]
I went back a bit and they added it to avoid confusion 😆 (https://github.com/w3c/html-aam/issues/55).  I have more of a DOM background so that line had the opposite effect one me.

**Comment by eps1lon** [comment]
Forgot about this after npm was down the day this issue got fixed. Should be available now with `dom-accessibility-api@^0.5.3`
